pkg/dao/repository.go

```diff
@@ -30,13 +34,32 @@ func DBErrorToApi(e error) error {
 }
 
 func (r repositoryDaoImpl) Create(newRepo api.RepositoryRequest) error {
-	newRepoConfig := models.RepositoryConfiguration{
-		AccountID: *newRepo.AccountID,
-		OrgID:     *newRepo.OrgID,
+	// Check arguments
+	if newRepo.URL == nil || *newRepo.URL == "" {
+		return fmt.Errorf("URL cannot be nil nor empty string")
+	}
+	if newRepo.OrgID == nil || *newRepo.OrgID == "" {
+		return fmt.Errorf("OrgID cannot be nil nor empty string")
+	}
+	if newRepo.AccountID == nil || *newRepo.AccountID == "" {
+		return fmt.Errorf("AccountID cannot be nil nor empty string")
+	}
```

Comment:
you can do validation in the model like this:  https://github.com/content-services/content-sources-backend/blob/main/pkg/models/repository_configuration.go#L32-L47

Reply:
thanks! updated